### ISSUE TITLE
Javascript API does not support extending rulesets

### ISSUE THREAD

**Issue by Jaredk3nt** [stoplightio/spectral#864]
**Describe the bug**
I have a ruleset that extends the oas2 ruleset. This ruleset works perfectly from the command line but when I try to port that into my code with the JS API I cannot extend the oas2 rules. ~In the CLI the `RulesetProcessor` it is checking for `ruleset.extends` and processing those in turn. However neither `spectral.loadRuleset()` or `spectral.setRuleset()` take into account extends. This is made worse by the fact that there is a `spectral.mergeRules()` function that could be used to merge in my own rules if I wanted to take all of the rules from oas2, but there isn't an equivalent for functions to be able to add my custom functions.~

**To Reproduce**

Try to lint a ruleset with an extends using the JS API. Here is the one I am using (without my custom functions) https://gist.github.com/Jaredk3nt/6cf890fd26a77fced3dd602f11607c0e

**Expected behavior**
~There should be some way to extend a ruleset, either automatically or manually.~

This was caused by me misreading the code and the docs not really having a good example. I was able to get my extended ruleset to work in the JS API by using loadRuleset:

```js
const spectral = new Spectral();
spectral.registerFormat("oas2", isOpenApiv2);

function validate(spec) {
  return spectral.loadRuleset(__dirname + "/my-rules.json").then(() => {
    return spectral
      .run({
        getLocationForJsonPath,
        parsed: parseWithPointers(spec)
      })
      .then(console.log);
  });
}
```

Hopefully this example will save someone else!

**Comment by Jaredk3nt** [comment]
Actually it looks like I misunderstood what `mergeRules` does, it looks like it doesn't actually allow you to add rules but instead update or override existing rules so that wouldn't allow me to add my custom rules to the oas2 ones anyways.
